nixos/modules/services/system/dbus.nix

```diff
@@ -122,6 +141,30 @@ in
 
     systemd.services.dbus.restartTriggers = [ configDir ];
 
+    systemd.user = mkIf config.systemd.user.dbus.enable {
+      services.dbus = {
+        description = "D-Bus User Message Bus";
+        requires = [ "dbus.socket" ];
```

Comment:
FWIW [upstream](https://cgit.freedesktop.org/dbus/dbus/tree/bus/systemd-user/dbus.service.in) has no `After = dbus.socket` and is (apart from the `wantedBy` settings) identical to this.